implementations/c/ockam/channel/channel_impl.c

```diff
@@ -0,0 +1,297 @@
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <ctype.h>
+#include "ockam/syslog.h"
+#include "ockam/memory.h"
+#include "ockam/key_agreement.h"
+#include "ockam/key_agreement/xx.h"
+#include "ockam/transport.h"
+#include "ockam/io/impl.h"
+#include "ockam/channel.h"
+#include "channel_impl.h"
+#include "ockam/codec.h"
+
+ockam_memory_t* gp_ockam_channel_memory = NULL;
```

Comment:
I would avoid having global variables. Each time channel instance is initiated this variable is overridden which is
1. Not obvious at all
2. Makes impossible using different memory implementations, which then makes all the complexity of having separate memory module with dispatch pointless, because when you have only 1 implementation at a time there are easier ways to achieve that.